gradle/test-config.gradle

```diff
@@ -0,0 +1,70 @@
+// unit/integration tests configurations
+//
+// https://www.petrikainulainen.net/programming/gradle/getting-started-with-gradle-integration-testing/
+// https://www.michael-bull.com/blog/2016/06/04/separating-integration-and-unit-tests-with-gradle
+// https://discuss.gradle.org/t/separate-execution-for-java-unit-and-integration-tests/8713
+
+final testsRootDir = 'src/test'
+final testsCommonsDir = "${testsRootDir}/common"
+final testsUnitDir = "${testsRootDir}/unit"
+final testsIntegrationDir = "${testsRootDir}/integration"
+
+sourceSets {
+    testCommon {
+        java {
+            srcDirs = ["$testsCommonsDir/java"]
+            compileClasspath += main.output
+            runtimeClasspath += main.output
+        }
+        resources.srcDirs = ["$testsCommonsDir/resources"]
+    }
+
+    test {
+        java {
+            srcDirs = ["$testsUnitDir/java"]
+            compileClasspath += main.output + testCommon.output
+            runtimeClasspath += main.output + testCommon.output
+        }
+        resources.srcDirs = ["$testsUnitDir/resources"]
+    }
+
+    testIntegration {
+        java {
+            srcDirs = ["$testsIntegrationDir/java"]
+            compileClasspath += main.output + testCommon.output
+            runtimeClasspath += main.output + testCommon.output
+        }
+        resources.srcDirs = ["$testsIntegrationDir/resources"]
+    }
+}
+
+configurations {
+    testCommonCompile.extendsFrom compile
+    testCommonRuntime.extendsFrom runtime
```

Comment:
you can have these in two lines as a one liner:
````groovy
testCommonCompile.extendsFrom compile, runtime
````

Reply:
i think `testCommonRuntime` also should be configured to be dependent on `runtime`. Not sure mixing compile and runtime configurations is a good idea, i'd keep them separated

Reply:
oh, my mistake. I thought they are both `testCommonCompile`. Please ignore it.